### DISCUSSION TITLE
General ideas

### DISCUSSION THREAD

**Discussion by float34** [microsoft/WindowsAppSDK#1943] [Ideas]
There are several problems I want to highlight and suggest my ideas.

1. Microsoft has created lots of frameworks over the years, and most of them are not deprecated. It became so confusing that even experienced developers struggle to sort this out and understand what technology to use for the new project.
**Idea:** Create a visual representation of all current technologies/stacks/SDKs (matrix, flow chart, graph, whatever) which can be used as a reference for any developer, coming from any technology/skill/experience.
**Motivation:** Windows is still the biggest development platform in the world, but new developers are so confused that they decide to switch to other platforms/stacks (Android/iOS).

2. WindowsAppSDK - a new boy in town. While it has potential it still feels like a hobby project and not something that you can invest in the long-term.
**Ideas:** 
a) Dedicate more resources to the dev team (Yeah I know you cannot get developers out of the thin air).
b) Provide more examples on how to use it.
c) Use it yourselves everywhere in Windows and its apps, even if it will be limited/with bugs in the beginning. **This is crucial for adoption.**
**Motivation:** As the new technology it needs to be embraced by developers. But Windows developers had enough bitter pills already, seriously. So many technologies were buried by Microsoft. Isn't it enough? Show that you taking this one seriously. Otherwise, no one will use it, because all will leave for other platforms.

3. Establish a proper communication channel.
**Idea:** Create a **Discord** channel with actual Windows developers easily reachable. Like **DirectX** channel.
**Motivation:** As you push the new **WindowsAppSDK** tech you probably want feedback, ideas, reports as quickly as possible. Having a **Reddit AMA** once in 6 month is laughable, really. Are you dedicated or not? Spend **1hr** a day to review and respond to messages in Discord, it will immediately give you the idea what's going on with your technology's perception/usage. 

4. Collect proposals differently, not with "productboard".
**Idea:** Use other tool to collect proposals.
**Motivation:** Productboard seems to not show all proposed ideas. Do you want to know what **WE WANT**, or you simply adding another useless filter for the people's ideas? Why opensourcing if you don't want to listen to actual customers? Please reconsider.

5. And please change the "Q&A" icon from praying hands to a question mark. You are not the gods, and we are not praying. This is humiliating.

Thank you for your attention!

**Comment by wjk** [comment]
This isn’t really an answer, but I can’t post my comments otherwise.

To me, productboard is very much a way for Microsoft to be able to announce they are developing Windows App SDK “per the wishes of the community” and not technically lie without actually addressing any of the countless problems the community has actually described. I consider this rather disingenuous public-relations tactic yet another public slight against our community, both as AppSDK users and as open-source developers in general. It needs to go; GitHub Issues and Discussions are more than enough.

The idea of starting a Discord server is just another attempt to _get you to listen to us at all_; an actual Discord server is completely unnecessary, and as far as I am concerned operating one would be a slight against our privacy. I understand that anything said on Discord may be fair game for them to sell, and I’d rather not have to discuss my potentially confidential code in such an environment. What we say here belongs to us.

Regarding that emoji, I think it is intended to look like two hands slapping each other in success. It’s a stretch for me as well, and I agree it should be removed.

**Comment by BenJKuhn** [comment]
Thank you for the detailed feedback and candor. Everything you've mentioned touches on topics that we've discussed amongst ourselves as well. We're continuing to evolve how we manage our work and engage with the community. Although It's going to take time to find the right answers to each of these, this type of input is helpful and appreciated. We'll keep listening, evolving, and trying new things. Some will work better than others, and I'm certain we won't please everyone.

Productboard has been an interesting experiment for us. We've been using it both for community engagement, planning, and progress tracking on major themes in ways that are difficult to express & track in github. It also allows us to track work we're doing in the open alongside work that we're not ready to disclose publicly, something we can't do in github. Most of us spent most of our time before working on Windows App SDK using ADO for project tracking, but that's completely closed. We'd hoped productboard would let us strike a better balance of being as open as possible while still keeping our major planning and tracking of both public & non-public information in a single view.

In short, we tend to perceive it differently from what you describe. I'm not sure if that disconnect is more about messaging or execution (probably a little of both), but either way it sounds like we have some work to do.

As for the Q&A icon, that's an easy one: I don't know who set it or what the motivation was but based on your input, it's not sending the intended message. I went ahead and changed it to the red phone for now. (FYI @btueffers)

Ben
